src/Promitor.Core.Scraping/AzureMonitorScraper.cs

```diff
@@ -73,6 +82,72 @@ protected override async Task<ScrapeResult> ScrapeResourceAsync(string subscript
             return new ScrapeResult(subscriptionId, scrapeDefinition.ResourceGroupName, resourceDefinition.ResourceName, resourceUri, finalMetricValues, metricLabels);
         }
 
+        protected override async Task<List<ScrapeResult>> BatchScrapeResourceAsync(string subscriptionId, BatchScrapeDefinition<IAzureResourceDefinition> batchScrapeDefinition, PromitorMetricAggregationType aggregationType, TimeSpan aggregationInterval)
+        {
+            Guard.NotNull(batchScrapeDefinition, nameof(batchScrapeDefinition));
+            Guard.NotLessThan(batchScrapeDefinition.ScrapeDefinitions.Count(), 1, nameof(batchScrapeDefinition));
+            Guard.NotNull(batchScrapeDefinition.ScrapeDefinitionBatchProperties.AzureMetricConfiguration, nameof(batchScrapeDefinition.ScrapeDefinitionBatchProperties.AzureMetricConfiguration));
+
+            var metricName = batchScrapeDefinition.ScrapeDefinitionBatchProperties.AzureMetricConfiguration.MetricName;
+
+            // Build list of resource URIs based on definitions in the batch 
+            var resourceUriList = new List<string>();
+            foreach (ScrapeDefinition<IAzureResourceDefinition> scrapeDefinition in batchScrapeDefinition.ScrapeDefinitions)
+            {
+                var resourceUri = BuildResourceUri(subscriptionId, scrapeDefinition, (TResourceDefinition) scrapeDefinition.Resource);
+                resourceUriList.Add(resourceUri);
+                // cache resource info 
+                if (!_resourceDefinitions.ContainsKey(resourceUri))
+                {
+                    var resourceDefinitionToCache = new AzureResourceDefinition
+                    (
+                        resourceType: scrapeDefinition.Resource.ResourceType, 
+                        resourceGroupName:  scrapeDefinition.ResourceGroupName, 
+                        subscriptionId: scrapeDefinition.SubscriptionId, 
+                        resourceName: scrapeDefinition.Resource.ResourceName
+                    ); // the resource definition attached is missing some attributes, filling them in here
+                    Logger.LogWarning("Caching resource group {Group} for {ResourceId}",  resourceDefinitionToCache.ResourceGroupName, resourceUri);
+                    _resourceDefinitions.TryAdd(resourceUri, resourceDefinitionToCache);
```

Comment:
Instead of contains key, we can consider using addorupdate instead as well though